### ISSUE TITLE
[Other - Accessibility] Single-click destructive action. (00.00.5)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#73990]
#### VFS team name
Benefits Non-Disability

#### VFS product name
Pension 527EZ Digital Form

#### VFS feature name
Form Reactivation

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
No VA.gov Experience Standard for the issue found.

### Issue Details
On [Other service names](https://staging.va.gov/pension/application/527EZ/military/general), the Remove button removes the entry you're working on with a single click and no check to verify that the action is intended. There's also no confirmation message presented to screen reader users that the action has taken place, the only indication that it's happened is the fields disappearing from the screen.

This is not technically a violation of [WCAG 3.3.2](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html), since you get the opportunity for review. 

But it's definitely introducing an opportunity for users accidentally losing data --- and for assistive technology users, losing data and maybe not even realizing it happened.

This issue also occurs on:
- [VA medical centers](https://staging.va.gov/pension/application/527EZ/medical/history/va-treatment/medical-centers)
- [Federal medical facilities](https://staging.va.gov/pension/application/527EZ/medical/history/federal-treatment/medical-centers)
- [Current employment](https://staging.va.gov/pension/application/527EZ/employment/current/history)
- [Spouse’s former marriage](https://staging.va.gov/pension/application/527EZ/household/marital-status/spouse-marriages)
- [Dependent children](https://staging.va.gov/pension/application/527EZ/household/dependents/add)
- [Gross monthly income](https://staging.va.gov/pension/application/527EZ/financial/income-sources)
- [Add an unreimbursed care expense](https://staging.va.gov/pension/application/527EZ/financial/care-expenses/add)
- [Add a medical or other unreimbursed expense](https://staging.va.gov/pension/application/527EZ/financial/medical-expenses/add)

### VA.gov Experience Standard
[Category Number 00, Issue Number 00](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Platform Recommendation
Two recommended changes:

1. Some kind of status message that receives focus when an issue has been removed, similar to the background-only alerts in use when changing contact information. If a visible status message isn't practical, at least some aria-live update should be announced to screen reader users.
2. Some kind of intermediate confirmation step between a user clicking Remove and the issue actually being removed. I believe the team working on the Supplemental Claims Form 20-0995 is exploring using a modal and intends to contribute back to the pattern in the design system.

See #56082 for how another team addressed this.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by katherine-fung** [comment]
I'm providing content for the confirmation messages for the team to use when it addresses this issue:

[Other service names](https://staging.va.gov/pension/application/527EZ/military/general)

Remove this name? 

We'll remove [First name] [Middle name] [Last name] [Suffix].

[BUTTON] Yes, remove this
[BUTTON] No, keep this

[VA medical centers](https://staging.va.gov/pension/application/527EZ/medical/history/va-treatment/medical-centers)

Remove this VA medical center? 

We'll remove **[name of VA medical center]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this


[Federal medical facilities](https://staging.va.gov/pension/application/527EZ/medical/history/federal-treatment/medical-centers)

Remove this federal medical facility? 

We'll remove **[name of federal medical facility]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

[Current employment](https://staging.va.gov/pension/application/527EZ/employment/current/history)

Remove this job? 

We'll remove **[job title]** from your current jobs. 

[BUTTON] Yes, remove this
[BUTTON] No, keep this

[Spouse’s former marriage](https://staging.va.gov/pension/application/527EZ/household/marital-status/spouse-marriages)

TBD

[Dependent children](https://staging.va.gov/pension/application/527EZ/household/dependents/add)

TBD

[Gross monthly income](https://staging.va.gov/pension/application/527EZ/financial/income-sources)

Remove this source of income? 

We'll remove **[income type]** from **[who pays this income field]**. 

[BUTTON] Yes, remove this
[BUTTON] No, keep this

[Add an unreimbursed care expense](https://staging.va.gov/pension/application/527EZ/financial/care-expenses/add)

Remove this care expense? 

We'll remove the expense for **[name of care provider]**. 

[BUTTON] Yes, remove this
[BUTTON] No, keep this

[Add a medical or other unreimbursed expense](https://staging.va.gov/pension/application/527EZ/financial/medical-expenses/add)

Remove this medical or other unreimbursed expense? 

We'll remove the expense for **[who receives this payment field]**. 

[BUTTON] Yes, remove this
[BUTTON] No, keep this

**Comment by katherine-fung** [comment]
When the team addresses this issue, please use this copy for the confirmation messages. Please bold the dynamic content in the confirmation messages. Thank you! 

Here's a visual reference for the style of the confirmation message: 

![Screen Shot 2024-01-22 at 3 27 38 PM](https://github.com/department-of-veterans-affairs/va.gov-team/assets/105367381/6e14f299-351e-423f-a01a-8a6926a6201d)


- [ ] [Other service names](https://staging.va.gov/pension/application/527EZ/military/general)

[header] Remove this name?

We'll remove **[First name] [Middle name] [Last name] [Suffix]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [VA medical centers](https://staging.va.gov/pension/application/527EZ/medical/history/va-treatment/medical-centers)

[header] Remove this VA medical center?

We'll remove **[name of VA medical center]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [Federal medical facilities](https://staging.va.gov/pension/application/527EZ/medical/history/federal-treatment/medical-centers)

[header] Remove this federal medical facility?

We'll remove **[name of federal medical facility]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [Current employment](https://staging.va.gov/pension/application/527EZ/employment/current/history)

[header] Remove this job?

We'll remove **[job title]** from your current jobs.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [Spouse’s former marriage](https://staging.va.gov/pension/application/527EZ/household/marital-status/spouse-marriages)

[header] Remove [name of former spouse]?

We'll remove **[name of former spouse]** as a former spouse of [name of current spouse].  

[BUTTON] Yes, remove them
[BUTTON] No, keep them

[Dependent children](https://staging.va.gov/pension/application/527EZ/household/dependents/add)

[header] Remove [name of child]? 

We'll remove **[name of child]** from your list of dependent children. 

[BUTTON] Yes, remove them
[BUTTON] No, keep them

- [ ] [Gross monthly income](https://staging.va.gov/pension/application/527EZ/financial/income-sources)

[header] Remove this source of income?

We'll remove **[income type]** from **[who pays this income field]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [Add an unreimbursed care expense](https://staging.va.gov/pension/application/527EZ/financial/care-expenses/add)

[header] Remove this care expense?

We'll remove the expense for **[name of care provider]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

- [ ] [Add a medical or other unreimbursed expense](https://staging.va.gov/pension/application/527EZ/financial/medical-expenses/add)

[header] Remove this medical or other unreimbursed expense?

We'll remove the expense for **[who receives this payment field]**.

[BUTTON] Yes, remove this
[BUTTON] No, keep this

**Comment by vbellissimo** [comment]
@ToddWebDev is this new confirmation message recently applied to the form a design system component? If so, do you know if applying @katherine-fung 's recommendations above would deviate from this?

cc @mayacoforma @gmaciolek

**Comment by katherine-fung** [comment]
I'm following up on Vicky's question above. If we can't implement the copy recommendations in my comment on January 26, can we use a confirmation message component that tells the user which item they're removing? Otherwise, it will be hard for assistive technology users to know exactly which item they're removing. 

This is the current confirmation message format in staging: 

<img width="1835" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOXFWQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4c72bc8cfadaa21551fe9c8cc943d231381fb86c/Screenshot%20%28435%29.png" alt="Screenshot (435).png" />



versus the example in the design system, that identifies the item in body copy

<img width="1280" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOXVWQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--569088d59d3d030827d85ce77bb3667d7fe85413/0955-modal-detail%20%281%29.png" alt="0955-modal-detail (1).png" />

**Comment by vbellissimo** [comment]
@gmaciolek @ToddWebDev do you have any insight on the comments above?
